rdt-open-reader/src/testCovidIndonesia/java/io/ona/rdt/robolectric/activity/OneScanActivityTest.java

```diff
@@ -111,6 +121,89 @@ private void verifyBarcodeResult(int viewId, String expectedResult) {
+        final String performPostScanActionsMethod = "performPostScanActions";
+        final String enableBatchScanField = "enableBatchScan";
+        final String dataArrayField = "dataArray";
+
+        Bundle bundle = new Bundle();
+        bundle.putString("status", "ok");
+        bundle.putString("barcodeText", barcodeText);
+        bundle.putBoolean(sensorTriggeredStr, sensorTriggered);
+        bundle.putString("productId", productId);
+        bundle.putString("lot", lot);
+        bundle.putString(expDateStr, expirationDate);
+        bundle.putString(serialNumberStr, serialNumber);
+        bundle.putString("additionalIdentifier", additionalIdentifier);
+        bundle.putString(sensorTriggeredStr, sensorNotTriggered);
+
+        // enable batch scan
+        ReflectionHelpers.setField(oneScanActivity, enableBatchScanField, true);
+
+        ReflectionHelpers.callInstanceMethod(oneScanActivity, performPostScanActionsMethod,
+                ReflectionHelpers.ClassParameter.from(Bundle.class, bundle));
+        ReflectionHelpers.callInstanceMethod(oneScanActivity, performPostScanActionsMethod,
+                ReflectionHelpers.ClassParameter.from(Bundle.class, bundle));
+
+        JSONArray dataArray = ReflectionHelpers.getField(oneScanActivity, dataArrayField);
+
+        Assert.assertEquals(2, dataArray.length());
+
+    }
+
+    @Test
+    public void testActivityResultShouldVerifyCorrectResultData() throws JSONException {
+        final String setResultAndFinishMethod = "setResultAndFinish";
+        final String enableBatchScanField = "enableBatchScan";
+        Bundle bundle = new Bundle();
+        bundle.putString("status", "ok");
+        bundle.putString("barcodeText", "test");
+        bundle.putBoolean("sensorTriggered", false);
+        bundle.putString("productId", "test-product");
+        bundle.putString("lot", "test-lot");
+        bundle.putString("expirationDate", TestUtils.getFormattedDateWithOffset(1, CovidRDTBarcodeFactory.RDT_BARCODE_EXPIRATION_DATE_FORMAT));
+        bundle.putString("serialNumber", "test-serial");
+        bundle.putString("additionalIdentifier", "test-additional-identifier");
+        final OneScanHelper.ScanResponse scanResponse = new OneScanHelper.ScanResponse(bundle);
+
+        ShadowActivity shadowOneScanActivity = Shadows.shadowOf(oneScanActivity);
+        ReflectionHelpers.setField(oneScanActivity, enableBatchScanField, true);
+        ReflectionHelpers.callInstanceMethod(oneScanActivity, setResultAndFinishMethod,
+                ReflectionHelpers.ClassParameter.from(OneScanHelper.ScanResponse.class, scanResponse));
+
+        String resultData = shadowOneScanActivity.getResultIntent().getStringExtra("data");
+        Assert.assertEquals(Activity.RESULT_OK, shadowOneScanActivity.getResultCode());
+        Assert.assertNotNull(resultData);
+        JSONObject resultObj = new JSONObject(resultData);
+        Assert.assertTrue(resultObj.has("scans"));
```

Comment:
Is it possible to verify the content of the scans to see if it matches what we expect?